api/spec/requests/ingests_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe "Projects API", type: :request do
+  include_context("authenticated request")
+  include_context("param helpers")
+
+  let!(:project) { FactoryBot.create(:project, draft: false) }
+  let(:project_id) { project.id }
+
+  let!(:attributes) {
+    {
+      source: markdown_source_params,
+      ingestionType: "epub"
+    }
+  }
+  let!(:valid_params) {
+    build_json_payload(attributes: attributes)
+  }
+
+  context "when a text ingest is triggered" do
+    let!(:path) { "/api/v1/projects/#{project_id}/ingest" }
+
+    before(:each) do
+      post path, headers: admin_headers, params: valid_params
+    end
+
+    it "creates and processes an ingestion" do
+      expect(response).to have_http_status(201)
```

Comment:
This is a strong personal preference, and I know there are other Manifold tests written like this so I get that it might seem ideal, but I strongly believe it can be a little confounding to have requests in specs happen in before blocks. The side effects should happen within a single example.

```ruby
it "creates an processes an ingestion" do
  expect do
    post path, headers: admin_headers, params: valid_params
  end.to change(Ingestion, :count).by(1)

  expect(response).to have_http_status(:created)
end
```

Rack / Rails also defines nice, human-readable symbols for status codes, so you can pass in things like `:ok` (200), `:created` (201), `:no_content` (204), `:bad_request` (400), `:unauthenticated` (401), `:forbidden` (403), `:not_found` (404), `:gone` (410), `:unprocessable_entity` (422), `:internal_server_error` (500).

You can still use the numbers if you prefer, I just find the symbols a little more readable.